src/gluonts/model/tpp/rmtpp/_network.py

```diff
@@ -0,0 +1,399 @@
+                )
+
+    def hybridize(self, active=True, **kwargs):
+        if active:
+            raise NotImplementedError(
+                "RMTPP blocks do not support hybridization"
+            )
+
+
+class RMTPPTrainingNetwork(RMTPPNetworkBase):
+
+    # noinspection PyMethodOverriding,PyPep8Naming,PyIncorrectDocstring
+    def hybrid_forward(
+        self, F, target: Tensor, valid_length: Tensor, **kwargs,
+    ) -> Tensor:
+        """
+        Computes the RMTPP negative log likelihood loss.
+
+        As the model is trained on past (resp. future) or context
+        (resp. prediction) "intervals" as opposed to fixed-length "sequences",
+        the number of data points available varies across observations. To
+        account for this, data is made available to the training network as a
+        "ragged" tensor. The number of valid entries in each sequence is provided
+        in a separate variable, :code:`xxx_valid_length`.
+
+        Parameters
+        ----------
+        F
+            MXNet backend.
+        target
+            Tensor with observations.
+            Shape: (batch_size, past_max_sequence_length, target_dim).
+        valid_length
+            The `valid_length` or number of valid entries in the past_target
+            Tensor. Shape: (batch_size,)
+
+        Returns
+        -------
+        Tensor
+            Loss tensor. Shape: (batch_size,).
+        """
+        if F is mx.sym:
+            raise ValueError(
+                "The RMTPP model currently doesn't support hybridization."
+            )
+
+        batch_size = target.shape[0]
+        # IMPORTANT: We add an additional zero at the end of each sequence
+        # It will be used to store the time until the end of the interval
+        target = F.concat(target, F.zeros([batch_size, 1, 2]), dim=1)
+        # (N, T + 1, 2)
+
+        ia_times, marks = F.split(
+            target, num_outputs=2, axis=-1
+        )  # inter-arrival times, marks
+        marks = marks.squeeze(axis=-1)  # (N, T + 1)
+
+        # TODO: Is this necessary?
+        valid_length = valid_length.reshape(-1).astype(
+            ia_times.dtype
+        )  # make sure shape is (batch_size,)
+
+        if self.apply_log_to_rnn_inputs:
+            ia_times_input = ia_times.clip(1e-8, np.inf).log()
+        else:
+            ia_times_input = ia_times
+        rnn_input = F.concat(ia_times_input, self.embedding(marks), dim=-1)
+        rnn_output = self.rnn(rnn_input)  # (N, T + 1, H)
+
+        rnn_init_state = F.zeros([batch_size, 1, self.rnn_hidden_size])
+        history_emb = F.slice_axis(
+            F.concat(rnn_init_state, rnn_output, dim=1),
+            axis=1,
+            begin=0,
+            end=-1,
+        )  # (N, T + 1, H)
+
+        # Augment ia_times by adding the time remaining until interval_length
+        # Afterwards, each row of ia_times will sum up to interval_length
+        ia_times = ia_times.squeeze(-1)  # (N, T + 1)
```

Comment:
This complains for me, as it expects -1 to be a keyword argument. Did it work for you?